Add tests for matching

Svetuf opened the pull request.

Svetuf committed:
Add tests for matching

shkryab commented on source/matching/tests/matching_test.cpp:
```diff
@@ -0,0 +1,21 @@
+#include <opencv2/opencv.hpp>
+#include <opencv2/core/cvstd.hpp>
+
+#include "../include/stitching/matching/knn_matching.h"
```
Это другой проект и к нему прилинкована библиотека мачинга. Инклуд должен быть через <>

Svetuf committed:
Fix

shkryab commented on CMakeLists.txt:
```diff
@@ -8,11 +8,14 @@ set(CMAKE_CXX_STANDARD_REQUIRED ON)
 add_subdirectory(source/featuring)
 
 add_subdirectory(source/matching)
+add_subdirectory(source/matching/tests)
```
Тест добавлять в проекте source/matching

Svetuf committed:
Fix Cmake subdirectory's

shkryab commented on source/matching/tests/matching_test.cpp:
```diff
@@ -0,0 +1,21 @@
+#include <opencv2/opencv.hpp>
+#include <opencv2/core/cvstd.hpp>
+
+#include <../include/stitching/matching/knn_matching.h>
```
Путь должен быть не относительным.

shkryab committed:
Исправлены ошибки сборки

Структура IFeatures переименована
Библеотека featuring переделана в интерефейсную
Фичеренг orb и surf разнесен по разным библиотекам

shkryab committed:
Merge branch 'develop' into feature/test_matching_IR-19
